src/lib/icu/meson.build

```diff
@@ -18,8 +18,18 @@ if icu_dep.found()
     'Init.cxx',
   ]
 elif not get_option('iconv').disabled()
-  have_iconv = compiler.has_function('iconv', prefix : '#include <iconv.h>')
-  conf.set('HAVE_ICONV', have_iconv)
+  have_iconv = c_compiler.has_header_symbol('iconv.h', 'iconv')
+  if have_iconv
+    libiconv = c_compiler.find_library('iconv')
```

Comment:
Does this work when there's no libiconv?

Reply:
If there's no libiconv it fails and moves on to the libc check.

Reply:
Not here:
```
Header <iconv.h> has symbol "iconv" : YES 

../../src/lib/icu/meson.build:23:4: ERROR: C shared or static library 'iconv' not found
```

Reply:
Hmm need to add required : false looks like.